geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/stats/GeoMesaMetadataStats.scala

```diff
@@ -212,6 +224,13 @@ class GeoMesaMetadataStats(val ds: AccumuloDataStore, statsTable: String, genera
 
   override def clearStats(sft: SimpleFeatureType): Unit = metadata.delete(sft.getTypeName)
 
+  override def close(): Unit = {
+    running.set(false)
+    if (scheduledCompaction != null) {
+      scheduledCompaction.cancel(false)
+    }
```

Comment:
Hmm.... if I'm understanding correctly, running.set(false) will start a race between 1) scheduledCompaction to set to null and 2) getting into the if block....

Seems like one could see either A) an NPE on 230 (after the check passes, the other thread nulls out scheduledCompaction) or B) the check fails and the compactions aren't canceled properly.

This seems like a problem in a web container situation where dispose a datastore could be common.